rust/feature-flags/src/flag_matching.rs

```diff
@@ -132,90 +288,144 @@ impl FeatureFlagMatcher {
-                return Ok(person_overrides.clone());
-            }
+        if let Some(overrides) = self.can_compute_property_overrides_locally(
+            &self.person_property_overrides,
+            flag_property_filters,
+        ) {
+            return Ok(overrides);
         }
 
         // If we don't prefer the overrides (they're either not present, don't contain enough properties to evaluate the condition,
         // or contain a cohort property), fall back to getting properties from cache or DB
-        self.get_person_properties_from_cache_or_db(team_id, self.distinct_id.clone())
+        self.get_person_properties_from_in_memory_cache_or_db(team_id, self.distinct_id.clone())
             .await
     }
 
+    /// Check if all required properties are present in the overrides
+    /// and none of them are of type "cohort" – if so, return the overrides,
+    /// otherwise return None, because we can't locally compute cohort properties
+    fn can_compute_property_overrides_locally(
+        &self,
+        property_overrides: &Option<HashMap<String, Value>>,
+        property_filters: &[PropertyFilter],
+    ) -> Option<HashMap<String, Value>> {
+        property_overrides.as_ref().and_then(|person_overrides| {
+            let should_prefer_overrides = property_filters
+                .iter()
+                .all(|prop| person_overrides.contains_key(&prop.key) && prop.prop_type != "cohort");
+
+            if should_prefer_overrides {
+                Some(person_overrides.clone())
+            } else {
+                None
+            }
+        })
+    }
+
     fn all_properties_match(
         &self,
-        condition_properties: &[PropertyFilter],
+        flag_condition_properties: &[PropertyFilter],
         target_properties: &HashMap<String, Value>,
     ) -> bool {
-        condition_properties
+        flag_condition_properties
             .iter()
             .all(|property| match_property(property, target_properties, false).unwrap_or(false))
     }
 
-    pub fn hashed_identifier(&self, feature_flag: &FeatureFlag) -> Option<String> {
-        if feature_flag.get_group_type_index().is_none() {
-            // TODO: Use hash key overrides for experience continuity
-            Some(self.distinct_id.clone())
+    /// This function takes a feature flag and returns the hashed identifier for the flag.
+    /// If the flag has a group type index, it returns the group type name, otherwise it returns the distinct_id.
+    pub async fn hashed_identifier(&self, feature_flag: &FeatureFlag) -> Result<String, FlagError> {
+        if let Some(group_type_index) = feature_flag.get_group_type_index() {
+            let indexes_to_names = self
+                .flag_matcher_cache
+                .group_type_index_to_group_type_map()
+                .await?;
+            Ok(indexes_to_names
+                .get(&group_type_index)
+                .cloned()
+                .unwrap_or_default())
```

Comment:
Probably worth a comment here, but this returns `""` if there's no group type <-> group index map.  That's `default()` in the context of a string type.